service/src/lib.rs

```diff
@@ -391,6 +393,15 @@ pub fn new_full<RuntimeApi, Executor>(
 
 		polkadot_network_service.register_availability_store(availability_store.clone());
 
+		let execution_mode = if test_mode {
```

Comment:
```suggestion
		let execution_mode = if !test_mode {
```

Reply:
Wait what. Why? The whole point was to get this on the worker args for the tests. Am I missing something?

Reply:
See my other comment, this makes no sense. In test mode you don't want to start the external worker and do it as thread instead. You mixed the concepts here totally. You should read the code of the validation again that decides on whether to start a validation worker or to run it inside the process.

Reply:
Oh no it's fine I get it now!

So Local would be use by the integration test on cumulus because we don't need it to be interruptible (or any more sophisticated), just need it to run for a few seconds during the test and everything is stopped anyway (because tests have their own process in Rust which is great I think).

Remote is the currently supported case in polkadot (and polkadot's tests!) where a background tasks is ran by spawning a new child process.

ez pz